lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -213,26 +242,28 @@ def compile(node, options)
     config = nil
 
     benchmark(:notice, str) do
-      Puppet::Util::Profiler.profile(str, [:compiler, :compile, node.environment, node.name]) do
+      compile_type = checksum_type ? :static_compile : :compile
+      Puppet::Util::Profiler.profile(str, [:compiler, compile_type, node.environment, node.name]) do
```

Comment:
Since the inlining metrics aren't grouped under `:compiler, :static_compile` anymore, is there value in `:compiler, :static_compile` as distinct from `:compiler, :compile`? It seems like it could get weird if there are existing metrics that are grouped under `:compiler, :compile`.